packages/transformer/src/IModelExporter.ts

```diff
@@ -927,26 +927,27 @@ export class ChangedInstanceIds {
   private elementECClassIds = new Set<string>();
   private aspectECClassIds = new Set<string>();
   private relationshipECClassIds = new Set<string>();
+  private _ecClassIdsInitialized = false;
   private _db: IModelDb;
   public constructor(db: IModelDb) {
     this._db = db;
-    this.setupECClassIds();
   }
 
-  private setupECClassIds(): void {
-    const addECClassIdsToSet = (setToModify: Set<string>, baseClass: string) => {
-      this._db.withPreparedStatement(`SELECT ECInstanceId FROM ECDbMeta.ECClassDef where ECInstanceId IS (${baseClass})`, (stmt) => {
-        while (stmt.step() === DbResult.BE_SQLITE_ROW) {
-          setToModify.add(stmt.getValue(0).getId());
-        }
-      });
+  private async setupECClassIds(): Promise<void> {
+    const addECClassIdsToSet = async (setToModify: Set<string>, baseClass: string) => {
+      for await (const row of this._db.createQueryReader(`SELECT ECInstanceId FROM ECDbMeta.ECClassDef where ECInstanceId IS (${baseClass})`)) {
+        setToModify.add(row.ECInstanceId);
+      }
     };
-    addECClassIdsToSet(this.codeSpecECClassIds, "BisCore.CodeSpec");
-    addECClassIdsToSet(this.modelECClassIds, "BisCore.Model");
-    addECClassIdsToSet(this.elementECClassIds, "BisCore.Element");
-    addECClassIdsToSet(this.aspectECClassIds, "BisCore.ElementUniqueAspect");
-    addECClassIdsToSet(this.aspectECClassIds, "BisCore.ElementMultiAspect");
-    addECClassIdsToSet(this.relationshipECClassIds, "BisCore.ElementRefersToElements");
+    const promises = [];
+    promises.push(addECClassIdsToSet(this.codeSpecECClassIds, "BisCore.CodeSpec"));
+    promises.push(addECClassIdsToSet(this.modelECClassIds, "BisCore.Model"));
+    promises.push(addECClassIdsToSet(this.elementECClassIds, "BisCore.Element"));
+    promises.push(addECClassIdsToSet(this.aspectECClassIds, "BisCore.ElementUniqueAspect"));
+    promises.push(addECClassIdsToSet(this.aspectECClassIds, "BisCore.ElementMultiAspect"));
+    promises.push(addECClassIdsToSet(this.relationshipECClassIds, "BisCore.ElementRefersToElements"));
+    await Promise.all(promises);
```

Comment:
nit: not a fan of unnecessary mutable state
```suggestion
    const promises = [
      addECClassIdsToSet(this.codeSpecECClassIds, "BisCore.CodeSpec"),
      addECClassIdsToSet(this.modelECClassIds, "BisCore.Model"),
      addECClassIdsToSet(this.elementECClassIds, "BisCore.Element"),
      addECClassIdsToSet(this.aspectECClassIds, "BisCore.ElementUniqueAspect"),
      addECClassIdsToSet(this.aspectECClassIds, "BisCore.ElementMultiAspect"),
      addECClassIdsToSet(this.relationshipECClassIds, "BisCore.ElementRefersToElements"),
    ];
    await Promise.all(promises);
```